tflog/provider_test.go

```diff
@@ -1293,6 +1633,57 @@ func TestMaskMessageStrings(t *testing.T) {
 	}
 }
 
+// Reference: https://github.com/hashicorp/terraform-plugin-log/issues/131
+func TestMaskMessageStrings_NewContext(t *testing.T) {
+	t.Parallel()
+
+	var outputBuffer bytes.Buffer
+
+	originalCtx := context.Background()
+	originalCtx = loggertest.ProviderRoot(originalCtx, &outputBuffer)
+	originalCtx = tflog.MaskMessageStrings(originalCtx, "original")
+
+	newCtx := tflog.MaskMessageStrings(originalCtx, "new")
+
+	tflog.Trace(originalCtx, "original should be masked")
+	tflog.Trace(originalCtx, "new should be preserved")
+	tflog.Trace(newCtx, "new should be masked")
```

Comment:
Is it worth using different log levels for `originalCtx` and `newCtx` in order to be able to differentiate between them in the `expectedOutput`?